src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

```diff
@@ -13,39 +15,55 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer;
 // but we did not want to migrate them all at once
 public class LoggerAdapter : IRazorLogger
 {
-    private readonly ILogger _logger;
+    private readonly IEnumerable<ILogger> _loggers;
     private readonly ITelemetryReporter? _telemetryReporter;
 
-    public LoggerAdapter(ILogger logger, ITelemetryReporter? telemetryReporter)
+    public LoggerAdapter(IEnumerable<ILogger> loggers, ITelemetryReporter? telemetryReporter)
```

Comment:
I needed the `OutputWindowLogger` in this PR because for whatever reason LogHub logs were either not being written or not being collected, and the output logger (which we used to have) presented a more reliable way of capturing logs that were being hidden. This was needed to find a couple of the issues which this PR hopes to resolve, and splitting it out now doesn't seem particularly helpful.

I'll see about passing the `ITelemetryReporter` as an `ILogger`.

Reply:
Passing the `ITelemetryReporter` as an `ILogger` was more complicated than it seemed at first glance. I filed https://github.com/dotnet/razor/issues/8125 to follow-up with this. The closer look convinced me it definitely has value because the behavior of our fault logging is currently owned by `LoggerAdapter` instead of by `TelemetryLogger` which it might more naturally be looked for.

Reply:
Note: We don't want to send *every* log message to the telemetry reporter, just errors and exceptions. So, it might be best to keep the telemetry reporter separate. cc @ryzngard

Reply:
I would have thought this was up to the `TelemetryLogger` to decide, by ignoring specific log levels, but that might be an incorrect assumption

Reply:
That's probably true. However, it also might make it a bit non-obvious that we're sending messages to telemetry. Just pointing out that abstracting here might make telemetry logging a bit subtle and it might be better to be explicit. No judgement from me! Just pointing it out.